app/src/main/java/org/jellyfin/mobile/player/PlayerFragment.kt

```diff
@@ -92,48 +97,57 @@ class PlayerActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        playerBinding = ActivityPlayerBinding.inflate(layoutInflater)
-        setContentView(playerBinding.root)
-        playerControlsBinding = ExoPlayerControlViewBinding.bind(findViewById(R.id.player_controls))
 
-        // Handle system window insets
-        ViewCompat.setOnApplyWindowInsetsListener(window.decorView) { _, insets ->
-            playerControlsView.updatePadding(left = insets.systemWindowInsetLeft, right = insets.systemWindowInsetRight)
-            playerOverlay.updatePadding(left = insets.systemWindowInsetLeft, right = insets.systemWindowInsetRight)
-            insets
+        // Set orientation to landscape and enable fullscreen initially, set status bar color
+        with(requireActivity()) {
+            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+            enableFullscreen()
+            window.statusBarColor = Color.BLACK
         }
 
         // Observe ViewModel
         viewModel.player.observe(this) { player ->
             playerView.player = player
-            if (player == null) finish()
+            if (player == null) parentFragmentManager.popBackStack()
         }
         viewModel.playerState.observe(this) { playerState ->
             val isPlaying = viewModel.playerOrNull?.isPlaying == true
+            val window = requireActivity().window
             if (isPlaying) {
                 window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
             } else {
                 window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
             }
             if (playerState == Player.STATE_ENDED) {
-                finish()
+                parentFragmentManager.popBackStack()
                 return@observe
             }
             loadingIndicator.isVisible = playerState == Player.STATE_BUFFERING
         }
         viewModel.mediaSourceManager.jellyfinMediaSource.observe(this) { jellyfinMediaSource ->
-            playbackMenus.onItemChanged(jellyfinMediaSource)
             titleTextView.text = jellyfinMediaSource.title
+            playbackMenus?.onItemChanged(jellyfinMediaSource)
         }
 
-        // Disable controller in PiP
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && isInPictureInPictureMode) {
-            playerView.useController = false
-        }
+        // Handle intent
+        viewModel.mediaSourceManager.handleArguments(requireArguments())
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
```

Comment:
```suggestion
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
```

If I remember correctly you can make the signature slightly stricter when overriding

Reply:
Ah yeah, good point. Thanks for spotting this :+1: